editor-components/slider/slider10/slider10.tsx

```diff
@@ -0,0 +1,470 @@
+      displayer: "Next Slide Button",
+      value: "FaArrowRight"
+    });
+  }
+
+  getName(): string {
+    return "Slider 10";
+  }
+
+  sliderNext = () => {
+    const updatedItems = this.getComponentState("slider-items").map((item: SliderItem) => {
+      if (item.index === this.getComponentState("slider-items").length - 1)
+        return {
+          ...item,
+          index: 0
+        }
+      else
+        return {
+          ...item,
+          index: item.index + 1
+        }
+    })
+    this.setComponentState("slider-items", updatedItems)
+  }
+
+  sliderPrev = () => {
+    const updatedItems = this.getComponentState("slider-items").map((index: SliderItem) => {
+      if (index.index === 0)
+        return {
+          ...index,
+          index: this.getComponentState("slider-items").length - 1
+        }
+      else
+        return {
+          ...index,
+          index: index.index - 1
+        }
+    })
+    this.setComponentState("slider-items", updatedItems)
+  }
+
+  render() {
+    const temp = this.getComponentState("slider-items-temp")
+
+    const props = this.castToObject<SliderItem[]>("slider-items").map((e: SliderItem) => {
+      return {
+        title: this.castToString(e.title),
+        image: e.image,
+        index: e.index
+      }
+    })
+
+    // console.log("temp", temp)
+    // console.log("props", props)
+
+    if (JSON.stringify(temp) !== JSON.stringify(props)) {
+      this.setComponentState("slider-items-temp", props);
+      this.setComponentState("slider-items", props);
+    }
+
+    return (
+      <div className={this.decorateCSS("container")}>
+        <div className={this.decorateCSS("max-content")}>
+          <div className={this.decorateCSS("slider-wrapper")}>
+            {this.getComponentState("slider-items").map((sliderItem: SliderItem, index: number) => {
+              const length = this.getComponentState("slider-items").length
+              const isItemTop = sliderItem.index === length - 1
+
+              return (
+                <div key={index} className={this.decorateCSS("slider-item")}
+                  style={{ opacity: isItemTop ? 1 : 0, zIndex: sliderItem.index, backgroundImage: `url('${sliderItem.image}')` }}>
+                  <h1
+                    style={{ marginTop: isItemTop ? 0 : "100px" }}
+                    className={this.decorateCSS("slider-item-title")}>
+                    {sliderItem.title}
+                  </h1>
+                  <div className={this.decorateCSS("slider-item-overlay")}></div>
+                </div>
+              )
+            })}
+            <footer className={this.decorateCSS("slider-footer")}>
+              <div className={this.decorateCSS("slider-footer-items")}>
+                <div className={this.decorateCSS("slider-footer-item")}>
+                  <img className={this.decorateCSS("slider-footer-item-image")}
+                    src={this.castToObject<FeaturedItem>("first-featured-item").image}
```

Comment:
Why can't you assign it to a variable? No need to castToObject every time. right?